datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachService.java

```diff
@@ -0,0 +1,132 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.service.OwnerAssignmentService;
+import com.sequenceiq.cloudbreak.common.service.TransactionService;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+
+/**
+ * Provides core logic and functionality for attaching or reattaching an SDX cluster.
+ */
+@Component
+public class SdxAttachService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxAttachService.class);
+
+    @Inject
+    private TransactionService transactionService;
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
```

Comment:
you should use `SdxService` instead

Reply:
I am open to doing this but based on one of my recent changes addressing your comments, I may also want to **delete** a cluster from the repository. This is not exposed by the `SdxService`.
What do you think I should do with this in mind?

Reply:
you should expose it using the service

Reply:
Done.